Mage.Sets/src/mage/cards/b/BronzebeakForagers.java

```diff
@@ -0,0 +1,141 @@
+                new ManaCostsImpl<>("{X}{W}")
+        );
+        dissolve_ability.setTargetAdjuster(BronzebeakForagerDissolveAdjuster.instance);
+        this.addAbility(dissolve_ability);
+    }
+
+    private BronzebeakForagers(final BronzebeakForagers card) {
+        super(card);
+    }
+
+    @Override
+    public BronzebeakForagers copy() {
+        return new BronzebeakForagers(this);
+    }
+}
+
+enum BronzebeakForagerExileAdjuster implements TargetAdjuster {
+    instance;
+
+    @Override
+    public void adjustTargets(Ability ability, Game game) {
+        ability.getTargets().clear();
+        for (UUID opponentId : game.getOpponents(ability.getControllerId())) {
+            Player opponent = game.getPlayer(opponentId);
+            if (opponent == null) {
+                continue;
+            }
+            FilterPermanent filter = new FilterPermanent("nonland permanent from opponent " + opponent.getLogName());
+            filter.add(new ControllerIdPredicate(opponentId));
+            filter.add(Predicates.not(CardType.LAND.getPredicate()));
+            TargetPermanent target = new TargetPermanent(0, 1, filter, false);
+            ability.addTarget(target);
+        }
+    }
+}
+
+// Based on Gelatinous Cube
+enum BronzebeakForagerDissolveAdjuster implements TargetAdjuster {
+    instance;
+
+    @Override
+    public void adjustTargets(Ability ability, Game game) {
+        ability.getTargets().clear();
+        int xValue = ability.getManaCostsToPay().getX();
+        FilterCard filter = new FilterCreatureCard("creature card with mana value " + xValue);
+        filter.add(new ManaValuePredicate(ComparisonType.EQUAL_TO, xValue));
+        ability.addTarget(new TargetCardInExile(filter, CardUtil.getExileZoneId(game, ability)));
+    }
+}
+
+// Based on Gelatinous Cube
+class BronzebeakForagerDissolveEffect extends OneShotEffect {
+
+    BronzebeakForagerDissolveEffect() {
+        super(Outcome.Benefit);
+        staticText = "put target creature card with mana value X exiled with {this} into its owner's graveyard";
+    }
+
+    private BronzebeakForagerDissolveEffect(final BronzebeakForagerDissolveEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public BronzebeakForagerDissolveEffect copy() {
+        return new BronzebeakForagerDissolveEffect(this);
+    }
+
+    // Lifegain based on Asmodeus the Archfiend
+    @Override
+    public boolean apply(Game game, Ability source) {
+        int xValue = source.getManaCostsToPay().getX();
+        Player controller = game.getPlayer(source.getControllerId());
+        Player player = game.getPlayer(source.getControllerId());
+        Card card = game.getCard(getTargetPointer().getFirst(game, source));
+        if (controller != null && player != null && card != null){
+            if (!player.moveCards(card, Zone.GRAVEYARD, source, game)) return false;
+            controller.gainLife(xValue, game, source);
```

Comment:
The life gain effect is independent from the card-moving effect. Rather than include it here, can use in the card class:

```java
dissolve_ability.addEffect(new GainLifeEffect(ManacostVariableValue.REGULAR));
```

(also a style nit, camel case for fields/methods: `dissolveAbility` rather than `dissolve_ability`)